packages/core/src/tracing/trace.ts

```diff
@@ -37,7 +38,7 @@ export function trace<T>(
 
   function finishAndSetSpan(): void {
     activeSpan && activeSpan.finish();
-    hub.getScope().setSpan(parentSpan);
+    scope.setSpan(parentSpan);
```

Comment:
@AbhiPrasad I was not 100% sure, but it is correct that we reset the span on the _current_ scope of the hub, not the one that we started it on? 🤔 Or possibly we should reset it on both? (same for below) - wydt?

Reply:
Thinking about this a bit we should probably fork the scope inside of this function otherwise it is a bit leaky...

Besides that I think the most correct behaviour is imo to only set the span on the scope that we started on.

Reply:
jup, agree! I'll "fix" this in a follow up PR.